### PR TITLE
4.1: multi-platform images

### PR DESCRIPTION
**Pull request by gg314**
### Out of scope?
* Decide status of `nodejs`
* Leave MLFlow server alone (for now)
* Large DL packages (pytorch, nlp things) are still conspicuously missing

### Updates
* Python: 3.10.8
* All Python packages
* Spark: Fixed broken install, update to match `ml-dev`
* DL: NVIDIA CUDA base image, update to match `ml-dev`
* Docker image tags: 4.1
* Github action version updated, `buildx` platforms expanded for `base`, `ml-dev`
* Documentation updated for easier maintenance

### Fixes
* #123
* #122  
* #94 
* #69 (improvement or not?)

### Removes
* `s3fs` (incompatible with current botocore, boto3, awscli tools). Use a CLI s3 browser like `s5cmd`?

### CODE REVIEW

File: orbyter-spark-dev/README.md [hunk 1]
```diff
@@ -1,37 +1,34 @@
 # manifoldai/orbyter-spark-dev:0.1
```
Comment (jmcarpenter2) [inline orbyter-spark-dev/README.md:1]:
```suggestion
# manifoldai/orbyter-spark-dev:4.1
```

File: orbyter-spark-dev/README.md [hunk 2]
```diff
@@ -1,37 +1,34 @@
...
 
-Dockerfile for [manifoldai/orbyter-spark-dev:0.1](https://hub.docker.com/r/manifoldai/orbyter-spark-dev)
+Dockerfile for [manifoldai/orbyter-spark-dev](https://hub.docker.com/r/manifoldai/orbyter-spark-dev)
 
 ## Usage
 
 To run a bash shell
 
-`docker run -it manifoldai/orbyter-spark-dev:0.1 bash`
+`docker run -it manifoldai/orbyter-spark-dev:latest bash`
```
Comment (jmcarpenter2) [inline orbyter-spark-dev/README.md:9]:
```suggestion
`docker run -it manifoldai/orbyter-spark-dev:4.1 bash`
```

**Review by jmcarpenter2 — COMMENTED** [review]
This is legit!  I am prolly not the closest to the codebase, so I don't have much to add but was curious. Just left a couple super tiny documentation changes to keep it consistent across files

**Comment by gg314** [comment]
Thanks @jmcarpenter2! Fixed these. I'll do a deep scour of the version tags before a final merge, too.

**Pull request merged**